### DISCUSSION TITLE
Path in Windows 10 ? And Linux ?

### DISCUSSION THREAD

**Discussion by pcar75** [jamulussoftware/jamulus#2013] [Bug Investigation]
Bonjour,
In Windows 10 , my 3.8.0 Jamulus server will show the message file but not will not create the log.

The 'Target' command line of the Desktop Shortcut (± 'launcher' in Linux terms) is 
`"C:\Program Files\Jamulus\Jamulus.exe" -s -F -T -w "Jamulus_SrvMsg.HTML" -l "Jamulus_SrvLog.log"

The 'Start in' value is 
`"C:\Program Files\Jamulus"

Is there any reason why 'Start in' works for one argument but not the other (or other arguments that can take a path or not) ?

Does the log only function if the server is declared public ?

BTW , the command-line options page of your site does not list the -w command !?

![image](https://user-images.githubusercontent.com/64435162/133137485-ab035ed6-d9da-4b56-90c8-c980b66f5fbb.png)

**Comment by gilgongo** [comment]
> BTW , the command-line options page of your site does not list the -w command !?

Yes, this has been corrected in the docs for the next release of the website, which should be out in the next few weeks once translations are done.

**Comment by henkdegroot** [comment]
Hi, are you running the "server" as administrator? As windows UAC might prevent jamulus to write into the program files folder, when it is not run as administrator. As the HTML welcome message is only read, that works without administrator rights.

**Reply by JaminShanti** [reply]
Not sure how secure the Jamulus Server is , not sure you want to be running a Open Source server that anyone in the world can connect to as administrator.  Pick somewhere a low level account can write to like, ~/jamulus.log  or %temp%/jamulus.log

**Comment by pcar75** [comment]
Thanks ! I tried with `-l "D:\Jamulus_SrvLog.log"`   and it works :-)
Confusingly, some programs will display a message saying that X operation requires admin permission and will ask "Continue ?", which does the trick !?
I will try to see if any programs does that or only MS programs ! ...

**Comment by pcar75** [comment]
Interesting ...

Current non-MS programs in user mode will not complete the operation but suggest another path (ex. C:\Users\...).  MS Programs in user mode will suggest "Continue" (as admin), "Ignore" and "Cancel".

In a Desktop Shortcut the 'Target' field is limited to 259 characters.
As my directory names are explicit and also include spaces, the whole server command was too long to insert in the target line, hence my reason to re-use the "Start in" directory already available.
So one can 
enable the DesktopShortcut>Properties>ShortcutTab>AdvancedProperties>Execute as administrator box
or 
in the Settings>System ... create a user environment variable (ex. Jamulus="D:\First level\Second level\Third level\Fourth level")  and modify the DesktopShortcut>Properties>ShortcutTab>Target line mentionned initially to `"C:\Program Files\Jamulus\Jamulus.exe" -s -F -T -w %Jamulus%\Jamulus_SrvMsg.HTML -l %Jamulus%\Jamulus_SrvLog.log`

**Reply by henkdegroot** [reply]
Instead of creating a new environment variable, you can use one of the existing once. Like %appadata% and then add \jamulus.

`"C:\Program Files\Jamulus\Jamulus.exe" -s -F -T -w %appdata%\Jamulus\Jamulus_SrvMsg.HTML -l %appdata%\Jamulus\Jamulus_SrvLog.log`

That will then use the folder which is also the default path for the settings files.

**Comment by pcar75** [comment]
RIght !  But since I don't want to swamp my Program disk (like forget to timely erase the log file)  , I  prefer to use my Data disk (i.e and own new Set variable).

Thanks @ henkdegroot

However ...

**Comment by pcar75** [comment]
Since I can boot to a Linux Jamulus USB Key, I wanted it's Jamulus server to use the same Welcome message and log to the same directory,
So I tried modifying the start-up script.  However the behavior is different !?

Linux Ubuntu 20.04 Bash script (jamulus-server-start) :
---------------------------------------------------------
(...)
JAMULUS_DAT="/media/jamulusos/Disque Dur/Media/My Music Tech and Other/Jamulus"
echo "Valeur de JAMULUS_DAT : "
echo $JAMULUS_DAT
sleep 10

/usr/bin/Jamulus --server --welcomemessage $JAMULUS_DAT/Jamulus_SrvMsg.HTML

JAMULUS_DAT=
echo "Valeur de JAMULUS_DAT : "
echo $JAMULUS_DAT
sleep 10
(...)

Results:
----------

/usr/bin$ jamulus-server-start
Valeur de JAMULUS_DAT : 
/media/jamulusos/Disque Dur/Media/My Music Tech and Other/Jamulus
- server mode chosen
- welcome message: /media/jamulusos/Disque
/usr/bin/Jamulus: Unknown option 'Dur/Media/My' -- use '--help' for help
Valeur de JAMULUS_DAT : 

jamulusos@jamulusos:/usr/bin$

Observations:
----------------
It seems that between the shell passing the argument to the program , the integrity of the argument is not maintained (i.e. terminate argument at a blank and does not know what the rest of it is).

Of course, a Linux shell lines not being as limited as a 'Target line ' in a Windows Desktop Shortcut, I could use the long names thus :
  /usr/bin/Jamulus --server --welcomemessage "/media/jamulusos/Disque Dur/Media/My Music Tech and Other/Jamulus/Jamulus_SrvMsg.HTML" --log "/media/jamulusos/Disque Dur/Media/My Music Tech and Other/Jamulus/Jamulus_SrvLogL.log"

But is there a better/more correct way to use environment variables and long directories names with spaces without losing anything ?

Thanks for any information.

**Reply by gilgongo** [reply]
What happens if you quote the variable, so: `echo "$JAMULUS_DAT"` - does that work?

**Comment by pcar75** [comment]
I have not tried `echo "$JAMULUS_DAT"` but with
/usr/bin/Jamulus --server --welcomemessage` "$JAMULUS_DAT"`/Jamulus_SrvMsg.HTML
it works great !
Thanks !
